linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analysis.scala

```diff
@@ -106,15 +105,13 @@ object Analysis {
     def displayName: String = {
       def typeRefDisplayName(tpe: TypeRef): String = tpe match {
         case primRef: PrimRef               => PrimRefDisplayNames(primRef)
-        case ClassRef(encodedName)          => decodeClassName(encodedName)
+        case ClassRef(encodedName)          => encodedName.nameString
         case ArrayTypeRef(base, dimensions) => "[" * dimensions + typeRefDisplayName(base)
       }
 
-      val (simpleName, paramTypes, resultType) =
-        ir.Definitions.decodeMethodName(encodedName)
-
-      simpleName + "(" + paramTypes.map(typeRefDisplayName).mkString(",") + ")" +
-      resultType.fold("")(typeRefDisplayName)
+      encodedName.simpleName.nameString + "(" +
+      encodedName.paramTypeRefs.map(typeRefDisplayName).mkString(",") + ")" +
+      encodedName.resultTypeRef.fold("")(typeRefDisplayName)
```

Comment:
The representation here is for human readers. The changes in the code in this PR preserve the output of `displayName` between before and after this commit.

The question should be: do we a different representation *in `MethodName.nameString` than this one? That's arguable, but for now I would say yes because `displayName` is ambiguous (in theory, there could exist a *class* named `int` on the classpath) whereas `nameString` is non-ambiguous.

Reply:
Yep, fair enough.